User Model field 수정

jong-seoung opened the pull request:
- [remove] : core관련 test file 삭제
- [feat] : User모델의 profile_image를 새로운 모델 UserImlage로 생성
- [feat]: swagger이 DEBUG TRUE일때만 작동하게 설정
- [test]: 회원가입시 유저 모델 생성 테스트 변경, 최초 로그인 입력값 테스트 추가
- [feat]: User모델에 job field명 jobs로 수정
- [feat]: User모델에 skills field 추가, Skill model 생성

Commit:
Cache clear

Commit:
remove: core관련 test file 삭제

Commit:
feat: User모델의 profile_image를 새로운 모델 UserImage로 생성

Commit:
feat(weheproject/urls.py): swagger가 DEBUG TRUE일때만 작동하게 설정

Commit:
test: 회원가입시 유저모델 생성 테스트 변경, 최초 로그인 입력값 테스트 추가

Commit:
feat: User 모델에 job filed 수정, Skill model & skills fied 추가

여러 개의 값들을 저장 해야 하는 경우를 생각해 줘야함
 - User model에 job field를 jobs로 변경, Job model을 생성 하여 다대다 관계로 연결
 - User model에 skills field 생성, Skill model을 생성후 다대다 관계로 연결

minuet1215 commented on user/models.py:
```diff
@@ -63,3 +73,9 @@ class Meta:
         swappable = 'AUTH_USER_MODEL'
+
+
+class UserImage(TimeStampedModel, models.Model):
```
User모델에서도 UserImage에 대한 정보를 식별하고 있어야 할 거 같아요.

minuet1215 replied:
UserImage가 User를 단방향 참조 하고 있다면, User 모델에서 UserImage에 대한 접근이 어려워요.

또한, 단적인 예로 임의의 User를 삭제시에 UserImage도 같이 삭제해야 하는데, UserImage가 자동으로 삭제되지 않을 수 있어요.

User 모델에도 UserImage를 인식할 수 있는 필드를 추가해야 할 것 같습니다.

minuet1215 commented on weheproject/urls.py:
```diff
@@ -14,30 +14,33 @@
+    path('api/v1/user/', include('user.urls')),
 ]
+
+if settings.DEBUG:
```
프로덕트 환경(`settings.DEBUG = False`)에서는 스웨거를 노출하면 안되니,
개발 환경일 때만 swagger를 노출시키도록 분리하는 건 좋은 것 같아요 👍👍👍

minuet1215 commented on user/models.py:
```diff
@@ -63,3 +73,9 @@ class Meta:
+class UserImage(TimeStampedModel, models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    file_size = models.BigIntegerField()
+    image_url = models.CharField(_('profile image'), max_length=255, blank=True, default="basic_profile_image")
```
user앱에 있는 모든 모델들에 대해 확장 가능성을 고려해서, 별도의 앱을 분리하는 고민과 함께,
모델 간의 단방향, 양방향에 대한 정의를 고민해봐야 할 것 같습니다

Commit:
perf: User모델과 UserImage모델 양방향 참조

User 모델에서 UserImage 모델을 참조할때, 직접 참조할 수 없는 경우 발생
User모델과 UserImage를 OneToOneField로 양뱡항 참조가 가능하게 수정

Commit:
chore: user모델의 career삭제

Commit:
test: user모델의 career관련 내용 삭제

Commit:
refactor: url패턴 변경

swagger url 패턴이 django 2.0 이상의 버전에서의 정의에 맞지않는 부분을 수정

Commit:
fix: User객체 생성시 오류 수정

User객체 생성시 user_image와 연결된 UserImage 필드가 생성이 안되는 오류 수정
SKill, UserImage의 이미지를 저장하는 CharFiled를 ImageField로 변경

Commit:
feat: login, logout 기능 swagger 연결

Commit:
feat: Access Token을 이용하여 소셜로그인 구현

Commit:
refactor: 소셜로그인 엑세스 토큰 발급

Commit:
feat: 소셜로그인 이후 jwt 토큰 발급

Commit:
chore
